src/xref2/link.ml

```diff
@@ -351,13 +398,50 @@ and open_ env parent = function
 let rec unit env t =
   let open Compilation_unit in
   let content =
-    match t.content with
-    | Module sg ->
-        let sg = signature env (t.id :> Id.Signature.t) sg in
-        Module sg
-    | Pack _ as p -> p
+    if t.Lang.Compilation_unit.linked || t.hidden then t.content
```

Comment:
Before, we never really wanted to link hidden modules.
Now, because of occurrence counting, we might want to link hidden modules: actually, with dune's wrapping, there chances that most occurrence will be in a hidden module.

However, this is indeed "extra caution", because when passing the module to link, this [code](https://github.com/ocaml/odoc/blob/master/src/odoc/odoc_link.ml#L11-L16) removes the content of the module if it's hidden!
I would say that the extra caution is not a problem.

Reply:
Alright, thanks